internal/controller/model_pod.go

```diff
@@ -71,9 +81,98 @@ func (g *PodLoadBalancerModelGenerator) findPort(subset *corev1.EndpointSubset,
 	return -1, errPortNotFoundInSubset
 }
 
+func buildIngressRule(ingress *networkingv1.NetworkPolicyIngressRule) (rule model.IngressRule) {
+	rule.Ports = make([]model.PolicyPort, 0, len(ingress.Ports))
+	for _, port := range ingress.Ports {
+		newPort := model.PolicyPort{
+			Protocol: *port.Protocol,
+			EndPort:  port.EndPort,
+		}
+		if port.Port != nil {
+			newPort.Port = &port.Port.IntVal
+		}
+		klog.V(1).Infof("Adding proto %s port %d to %d",
+			newPort.Protocol, newPort.Port, newPort.EndPort)
+		rule.Ports = append(rule.Ports, newPort)
+	}
+
+	rule.AllowedIPBlocks = make([]model.AllowedIPBlock, 0, len(ingress.From))
+	for _, from := range ingress.From {
+		if from.IPBlock == nil {
+			continue
+		}
+		newBlock := model.AllowedIPBlock{
+			Cidr: from.IPBlock.CIDR,
+		}
+		for _, except := range from.IPBlock.Except {
+			newBlock.Except = append(newBlock.Except, except)
+		}
+		klog.V(1).Infof("Adding block %s with %d excepts",
+			newBlock.Cidr, len(newBlock.Except))
+		rule.AllowedIPBlocks = append(rule.AllowedIPBlocks, newBlock)
+	}
+	return rule
+}
+
+func buildNetworkPolicy(in *networkingv1.NetworkPolicy) (policy model.NetworkPolicy) {
+	policy = model.NetworkPolicy{
+		Name:         in.Name,
+		IngressRules: make([]model.IngressRule, len(in.Spec.Ingress)),
+	}
+	for i, ingress := range in.Spec.Ingress {
+		klog.V(1).Infof("Processing policy ingress %#v", ingress)
+		policy.IngressRules[i] = buildIngressRule(&ingress)
+	}
+	return policy
+}
+
+func hasPolicyType(policy *networkingv1.NetworkPolicy, policyType networkingv1.PolicyType) bool {
+	for _, element := range policy.Spec.PolicyTypes {
+		if element == policyType {
+			return true
+		}
+	}
+	return false
+}
+
 func (g *PodLoadBalancerModelGenerator) GenerateModel(portAssignment map[string]string) (*model.LoadBalancer, error) {
 	result := &model.LoadBalancer{}
 
+	allPolicies, err := g.networkpolicies.List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+
+	networkPolicies := make([]model.NetworkPolicy, 0, len(allPolicies))
+	policyMap := map[string][]string{} // dest addr => ingress ipBlock
+	for _, pol := range allPolicies {
+		klog.Infof("Processing policy %s", pol.Name)
+		if !hasPolicyType(pol, "Ingress") {
```

Comment:
@horazont The only policies that are filtered out, are those that don't apply to Ingress. For all others, a NetworkPolicy and a PolicyAssignment is generated. buildNetworkPolicy() will filter out ingress rules that are not IPBlocks but will return a policy in any case. The policyMap does not care at all about what ingress rules the policy contains. Thus, there is no need for special handling namespaceSelectors and podSelectors. As soon as there is a policyAssignment, the default rule for the pod will be 'drop'.